### PR TITLE
Branch rename warning

### LINKED ISSUE #1480: Rename Branch polish
screen shot 2017-05-15 at 9 35 37 pm

1. We should use the ⚠️ icon like we do in Create Branch if the branch will be created with a different name from what they typed in. We should probably use the same font size as well.

screen shot 2017-05-15 at 9 35 46 pm

2. Could probably use a little padding between the "Will be created …" message and the textbox.

/cc @donokuda @niik

### PR DESCRIPTION
**Pull request by vbfox**
This fixes #1480 

And also change the behavior of the branch-rename when the name is all whitespace (It disable the button and display the warning, as the create-branch dialog is doing)

A few remarks:
* I let the code live in `create-branch`, but I can extract it as having a dialog referencing another feel strange (I was just wondering where?)
* Maybe extracting a `BranchNameInDialog` component (With both the textbox and the warning) would be better? Code is pretty small so maybe it's not worth it.
* 2 spaces were auto trimmed by vscode settings, should I revert that ?

2017-05-18 17_14_02-github desktop-dev

### CODE REVIEW

**Comment by vbfox** [comment]
(I'll fix the lint errors when back on a PC)

File: app/src/ui/lib/branch-names.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export function renderBranchNameWarning(proposedName: string, sanitizedName: string) {
+  if (/^\s+$/.test(proposedName)) {
+    return renderWarningMessage('Branch name cannot be empty')
+  } else if (proposedName !== sanitizedName) {
+    return renderWarningMessage(`Will be created as ${sanitizedName}`)
+  } else {
+    return null
+  }
+}
```
Comment (joshaber) [inline app/src/ui/lib/branch-names.tsx]: Nit: needs a newline at the end of the file.

File: app/src/ui/create-branch/index.tsx [hunk 2]
```diff
@@ -10,10 +10,10 @@ import { Button } from '../lib/button'
 import { LinkButton } from '../lib/link-button'
 import { ButtonGroup } from '../lib/button-group'
 import { Dialog, DialogError, DialogContent, DialogFooter } from '../dialog'
-import { Octicon, OcticonSymbol } from '../octicons'
 import { VerticalSegmentedControl } from '../lib/vertical-segmented-control'
 import { TipState, IUnbornRepository, IDetachedHead, IValidBranch } from '../../models/tip'
 import { assertNever } from '../../lib/fatal-error'
+import { renderBranchNameWarning } from '../lib/branch-names';
```
Comment (joshaber) [inline app/src/ui/create-branch/index.tsx]: At the risk of bike shedding, let's name the file something more specifically about warnings. `branch-name-warnings` maybe?

**Comment by joshaber** [comment]
Looks good, thanks @vbfox ✨

**Pull request merged**